tests/integration/sqlite_src_test.py

```diff
@@ -0,0 +1,73 @@
+"""SQLite Intake Source integration tests."""
+from __future__ import annotations
+
+import logging
+
+import intake
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+from intake_sqlite import (
+    SQLiteSource,
+    SQLiteSourceAutoPartition,
+    SQLiteSourceManualPartition,
+)
+
+# pytest imports this package last, so plugin is not auto-added
+intake.register_driver(name="sqlite", driver=SQLiteSource)
+intake.register_driver(name="sqlite_auto", driver=SQLiteSourceAutoPartition)
+intake.register_driver(name="sqlite_manual", driver=SQLiteSourceManualPartition)
+
+logger = logging.getLogger(__name__)
+
+
+def test_temp_db_fixture(temp_db: tuple[str, str, str], df1: pd.DataFrame) -> None:
+    """Make sure a direct read from the temp DB works."""
+    table, table_nopk, urlpath = temp_db
+    actual = pd.read_sql(table, f"sqlite:///{urlpath}", index_col="pk")
+    assert_frame_equal(df1, actual)
+
+
+def test_simple_src(temp_db: tuple[str, str, str], df1: pd.DataFrame) -> None:
+    """Test simple table read from the SQLite catalog."""
+    table, table_nopk, urlpath = temp_db
+    actual = SQLiteSource(urlpath, table, sql_kwargs=dict(index_col="pk")).read()
+    assert_frame_equal(df1, actual)
+
+
+def test_auto_src_partition(temp_db: tuple[str, str, str], df1: pd.DataFrame) -> None:
+    """Test automatic partitioning of table."""
```

Comment:
Did you figure out how it partitions the table?

Reply:
Not really, at least not in the case of the automatic partitioning. I adapted these test cases from the `intake-sql` test suite.

For the manual partitions you give it a list of `WHERE` clauses, which seems kind of janky -- like you could easily mess them up and have the same records showing up in more than one partition.

Also, because `dask` can't deal with multi-indexes, and the partitioning is done based on the index, we'll have to think a bit about how to use the partitioning if/when we get to tables that we want to read in their entirety, and are too large to work with in memory.

It looks like the behavior is really controlled by `dask.dataframe.read_sql_table`